Enhance targets.json with components

theamirocohen opened the pull request:
### Description
The enhancement is further specified [here](https://jira.arm.com/browse/IOTSTOR-777).
Due to the enhancement some boards failed the general block device tests for flashiap component,
The fails were due to boards containing inconsistent sector sizes.
The tests were modified but should be improved to address the problem.

### Pull request type
    [ ] Fix
    [ ] Refactor
    [X] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers
@ARMmbed/mbed-os-storage

0xc0170 commented on targets/targets.json:
```diff
@@ -4235,10 +4239,8 @@
-            "MBEDTLS_PSA_CRYPTO_C"
-        ],
+        "components_add": ["SD", "FLASHIAP"],
+        "macros_add": ["MBEDTLS_PSA_CRYPTO_C"],
```
Why changing styling for this line? should be separate commit if needed

0xc0170 commented on features/storage/TESTS/blockdevice/general_block_device/main.cpp:
```diff
@@ -195,6 +195,11 @@ void basic_erase_program_read_test(BlockDevice *block_device, bd_size_t block_si
     // Find a random block
     bd_addr_t block = (rand() * block_size) % (block_device->size());
 
+    // Flashiap boards with inconsistent sector size will to align with random start addresses
```
How is this related to json update? Separate commit?

0xc0170 reviewed (approved):
> The tests were modified but should be improved to address the problem.

Where is this issue tracked?

jeromecoutant commented on targets/targets.json:
```diff
@@ -2560,7 +2562,7 @@
     },
     "DISCO_F413ZH": {
-        "components_add": ["QSPIF"],
+        "components_add": ["QSPIF", "SD", "FLASHIAP"],
```
I don't think adding SD is correct
(on DISCO_F413ZH, SD connector is using SDIO not SPI)

theamirocohen replied:
Hi @jeromecoutant,
So how can I approach this? should I delete it?

jeromecoutant commented on targets/targets.json:
```diff
@@ -3791,7 +3795,7 @@
             "STM_EMAC"
         ],
-        "components_add": ["QSPIF"],
+        "components_add": ["QSPIF", "SD", "FLASHIAP"],
```
SD seems not correct ?

theamirocohen replied:
Hi @jeromecoutant,
How do you know it is not correct?

jeromecoutant replied:
Because board doesn't have any embedded SD connector over SPI

Commit:
Enhance targets.json with components

Adding storage components to boards

Commit:
Modifing general block device tests doe to targets enhancement

Due to targets enhancement some boards failed the general block device tests for flashiap component,
The fails were due to boards containing inconsistent sector sizes.
The tests were modified but should be improved to address the problem.
Rand() function issues were fixed.
